contracts/evmscript/EVMScriptRunner.sol

```diff
@@ -0,0 +1,60 @@
+pragma solidity ^0.4.18;
+
+import "../apps/AppStorage.sol";
+import "./IEVMScriptExecutor.sol";
+import "./IEVMScriptRegistry.sol";
+
+import "./ScriptHelpers.sol";
+
+
+contract EVMScriptRunner is AppStorage, EVMScriptRegistryConstants {
+    using ScriptHelpers for bytes;
+
+    function runScript(bytes script, bytes input, address[] blacklist) protectState internal returns (bytes output) {
+        // TODO: Too much data flying around, maybe extracting spec id here is cheaper
+        address executorAddr = getExecutor(script);
+        require(executorAddr != address(0));
+
+        bytes memory calldataArgs = script.encode(input, blacklist);
+        bytes4 sig = IEVMScriptExecutor(0).execScript.selector;
+
+        require(executorAddr.delegatecall(sig, calldataArgs));
+
+        return returnedDataDecoded();
+    }
+
+    function getExecutor(bytes script) public view returns (IEVMScriptExecutor) {
+        return IEVMScriptExecutor(getExecutorRegistry().getScriptExecutor(script));
+    }
+
+    // TODO: Internal
+    function getExecutorRegistry() internal view returns (IEVMScriptRegistry) {
+        address registryAddr = kernel.getApp(EVMSCRIPT_REGISTRY_APP);
+        return IEVMScriptRegistry(registryAddr);
```

Comment:
We should add a `require(registryAddr != address(0))` check here.

Reply:
Why? Call will fail regardless when trying to call a `IEVMScriptRegistry(0)`